migrations/challenges/sync-all-challenges.js

```
@@ -0,0 +1,40 @@
+import Bluebird from 'Bluebird';
+
+import { model as Challenges } from '../../website/server/models/challenge';
+import { model as User } from '../../website/server/models/user';
+
+async function syncChallengeToMembers (challenges) {
+  challenges.forEach(async function (challenge) {

```

this will still do all the challenges in parallel and it'll return at line 19 immediately because forEach doesn't work with await, it'll kickstart the async function and then return

---

So, what is wrong with that?

---

I guess if a user is in two challenges that would be an issue?

---

well the problem is that any error inside these async functions will get swallowed as they're not part of a chain with a `catch` statement or must be wrapped inside a try/catch `async function () { try {code} catch (e) {handle}` and also that some challenges are really big and the database would crash. Consider that the onboarding emails process only 10 users in parallel and the cpu on the database grows to more than 20%

---

and while you have a limit(10) in the query it'll start a new one as soon as the forEach has run without waiting for the first 10 challenges to be processed